### ISSUE TITLE
No rule to make target '.bin'

### ISSUE THREAD

**Issue by craftmaster1231** [libopencm3/libopencm3-template#16]
Got an error after cleaning some of code. 
What I did:
1) git clone and update submodules
2) removed `my-common-files/` dir
3) rewrote main.c to only contain empty main function.
4) rewrote Makefile, changing basic things.
5) Build lib in `libopencm3/` with `make`.
6) Build in `stm/`  (old my-project) and got an error.
```

  CC	main.c
  GENLNK  stm32f103c8t6
  LD	stm32
  LD	.elf
make: *** No rule to make target '.bin', needed by 'all'.  Stop.


```
Here is how my Makefile looks like:
```

PROJECT = stm32
BUILD_DIR = bin

CFILES = main.c

# TODO - you will need to edit these two lines!
DEVICE=stm32f103c8t6
OOCD_FILE = board/stm32f103c8t6discovery.cfg

# You shouldn't have to edit anything below here.
VPATH += $(SHARED_DIR)
INCLUDES += $(patsubst %,-I%, . $(SHARED_DIR))
OPENCM3_DIR=../libopencm3

include $(OPENCM3_DIR)/mk/genlink-config.mk
include ../rules.mk
include $(OPENCM3_DIR)/mk/genlink-rules.mk

```

Can suppose I'm not the only one getting this error. Getting this error on multiple devices and OS. Hope I'm just missing some basic knowledge and it is not a bug.

**Comment by karlp** [comment]
step 4 is kinda important.... :)

Are you sure that's the makefile you have now? not trimmed or edited or anything?  You've got _two_ "LD" lines, with a ".elf" by itself, that looks super suspicious.   I can't work out what else you've managed to do to create that though, but you've admitted that you've been rewriting themakefiles...

I've tried the following, based on your steps:
```
rm -rf my-common-files
mv my-project stm
```
Then, edited the `stm/my-project.c` to have just an empty stub to compile, instead of using the common code.
Then, edited the `stm/Makefile` to have your project name, and drop the shared files.

And this still works for me.

Try running make with V=1, so you can see the command output in more detail, ala:
```
$ make V=1 -C stm
make: Entering directory '/home/karlp/src/libopencm3-template/stm'
  CC	my-project.c
arm-none-eabi-gcc -Os -std=c99 -ggdb3 -mcpu=cortex-m3 -mthumb -msoft-float -fno-common -ffunction-sections -fdata-sections -Wextra -Wshadow -Wno-unused-variable -Wimplicit-function-declaration -Wredundant-decls -Wstrict-prototypes -Wmissing-prototypes  -MD -Wall -Wundef -I. -I. -I../libopencm3/include -I. -I. -I../libopencm3/include  -DSTM32F1 -DSTM32F103C8T6 -I../libopencm3/include -o bin/my-project.o -c my-project.c
  GENLNK  stm32f103c8t6
arm-none-eabi-gcc -E -mcpu=cortex-m3 -mthumb -msoft-float -DSTM32F1 -DSTM32F103C8T6 -D_ROM=64K -D_RAM=20K -D_ROM_OFF=0x08000000 -D_RAM_OFF=0x20000000 -P -E ../libopencm3/ld/linker.ld.S -o generated.stm32f103c8t6.ld
  LD	stm32.elf
arm-none-eabi-gcc -Tgenerated.stm32f103c8t6.ld -L../libopencm3/lib -nostartfiles -mcpu=cortex-m3 -mthumb -msoft-float -specs=nano.specs -Wl,--gc-sections -L../libopencm3/lib bin/my-project.o   -lopencm3_stm32f1 -Wl,--start-group -lc -lgcc -lnosys -Wl,--end-group -o stm32.elf
  OBJCOPY	stm32.bin
arm-none-eabi-objcopy -O binary  stm32.elf stm32.bin
make: Leaving directory '/home/karlp/src/libopencm3-template/stm'
```

I can get the same error as you by commenting out the "PROJECT" variable, but you seem to have it both set, and not-set.

**Comment by craftmaster1231** [comment]
Tried to reproduce error with list of commands to make it more clean for you. Ended up with perfectly working project with same structure. :))
After some time I've noticed that problem is in my project name. It has a SPACE symbol at the end like 'stm32 '. It was not sent to original post somehow. 

TY for your effort.
Now I established importance of indentation symbols in Makefiles and power of `:set syntax` in vim. =)
